runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpInfoBox.java

```diff
@@ -139,18 +139,18 @@
 		popupMenu.add(resetOthers);
 		popupMenu.add(pauseSkill);
 		popupMenu.add(canvasItem);
+		popupMenu.addPopupMenuListener(new SimplePopupMenuListener(e ->
```

Comment:
better (or well, simpler) would be to just use anonymous implementations here (we already do it below, no reason to overcomplicate things)

Reply:
I'm not sure I follow, but PopupMenuListener interface is not functional so you can't do straight lambda like on L145. Anonymous implementation expands to
```java
		popupMenu.addPopupMenuListener(new PopupMenuListener()
		{
			@Override
			public void popupMenuWillBecomeVisible(PopupMenuEvent popupMenuEvent)
			{				
			}

			@Override
			public void popupMenuWillBecomeInvisible(PopupMenuEvent popupMenuEvent)
			{
			}

			@Override
			public void popupMenuCanceled(PopupMenuEvent popupMenuEvent)
			{
			}
		});
```
Hence why `PopupMenuOpenListener` is a functional interface for the required action and `SimplePopupMenuListener` is a simple wrapper around entire 3 method interface.

Reply:
Is the implementation of all 3 methods inline better?

Reply:
yeah, that is exactly what I meant